portal-impl/src/com/liferay/portal/service/impl/CompanyLocalServiceImpl.java

```diff
@@ -333,9 +333,22 @@ public Company checkCompany(String webId, String mx)
 			if (defaultUser != null) {
 				if (!defaultUser.isAgreedToTermsOfUse()) {
 					defaultUser.setAgreedToTermsOfUse(true);
+				}
+
+				defaultUser.setLanguageId(companyDefaultLocale.toString());
```

Comment:
If you do this then any change done through Control Panel will be overwritten on start up (with whatever is in the properties), which is not what the admin will expect since it's not the case for all other properties.